packages/web-config-server/src/apiV1/measureData.js

```diff
@@ -80,18 +80,21 @@ const accessDeniedForMeasure = {
  * }
  */
 const buildMeasureData = (overlays, resultData) => {
-  const measureDataResponsesByMeasureId = resultData.reduce((dataResponse, current) => ({
+  const measureDataResponsesByMeasureCode = resultData.reduce((dataResponse, current) => ({
     ...dataResponse,
     ...current,
   }));
-  const measureDataResponses = overlays.map(({ id, dataElementCode }) => {
-    const { data } = measureDataResponsesByMeasureId[id];
-
-    return data.map(obj => {
-      const { [dataElementCode]: value, ...restData } = obj;
-      return { [id]: value, ...restData };
-    });
-  });
+  const measureDataResponses = overlays.map(
+    ({ code, data_builder_config: measureBuilderConfig }) => {
+      const { dataElementCode = 'value' } = measureBuilderConfig ?? {};
+      const { data } = measureDataResponsesByMeasureCode[code];
+
+      return data.map(obj => {
+        const { [dataElementCode]: value, ...restData } = obj;
+        return { [code]: value, ...restData };
```

Comment:
I found this section a bit hard to parse, but can't think of a better way - it's just quite clever with the dynamic destructuring that gets renamed to a new variable. Perhaps avoiding reuse of the word `value` would help a little
```suggestion
        const { [dataElementCode]: measureValue, ...restData } = obj;
        return { [code]: measureValue, ...restData };
```